datalad/distribution/publish.py

```diff
@@ -85,115 +86,230 @@ class Publish(Interface):
         with_data=Parameter(
             args=("--with-data",),
             doc="shell pattern",
-            action="store_true",
             constraints=EnsureListOf(string_types) | EnsureNone()),)
 
```

Comment:
I think this needs `nargs='*'` to work. Currently, it will just pass a single arg on to the Python function, and does not gather them in a list, making --with-data fail in all cases.